### ISSUE TITLE
`.detach()` in QuantumLayer.forward severs the gradient path into the encoder / input_proj

### ISSUE THREAD

**Issue by Kuo-TingKai** [ChiShengChen/Quantum_MNIST_Diffusion#2]
## Summary

`QuantumLayer.forward` calls `.detach()` on its input, which removes the quantum branch from the autograd graph on the input side. As a result no gradient flows back through the quantum layer into the encoder (or, in `v7`, into `input_proj`).

## v7 — `quantum_difussion_mnist_v7.py` / `quantum_difussion_pathmnist_v7.py`

```python
def forward(self, x):
    x = self.input_proj(x)          # nn.Linear(128, 16)
    x_cpu = x.detach().cpu()        # <-- graph is cut here
    results = [self.qlayer(sample) for sample in x_cpu]
    out = torch.stack(results).to(x.device)
    return self.output_proj(out)
```

`x_cpu` does not require grad, so `input_proj` receives **exactly zero gradient** for the entire run. Its weights stay at their Kaiming-uniform initialization for all 30 epochs. The 128→16 projection that feeds the circuit is therefore a fixed random projection that is never adapted to the task.

## v8 — `full_unet/quantum_diffusion_mnist_v8.py`

```python
def forward(self, x):
    x_cpu = x.detach().cpu()
    quantum_output = self.qlayer(x_cpu)
    ...
```

There is no `input_proj` here, so the cut lands directly between the U-Net bottleneck and the quantum layer. The channel gate `mid = mid * q_weight(...)` still changes the forward pass, but the encoder never receives any learning signal about *what input would make the quantum gate useful*. The gate is effectively driven by an unsupervised, uncontrolled statistic of the bottleneck.

## Why it matters for the paper

Together with #1 (the circuit's `weights` have identically zero gradient by construction), this means that in v7 the trainable content of the entire quantum branch is:

| component | trainable? | why |
|---|---|---|
| `input_proj` (128→16) | ❌ | cut by `.detach()` |
| `qlayer.weights` (3×16) | ❌ | circuit is diagonal + permutation only |
| `output_proj` (16→128) | ✅ | downstream of the cut |

So `use_quantum=True` adds exactly one trainable `nn.Linear(16, 128)` on top of a fixed nonlinear feature map, relative to `use_quantum=False`.

## Note on the likely intent

I assume `.detach().cpu()` was added because `default.qubit` execution was easier to keep on CPU. That is fixable without detaching:

```python
# keeps the graph, just moves the tensor
x_cpu = x.to('cpu')
out = self.qlayer(x_cpu).to(x.device)
```

`qml.qnn.TorchLayer` with `diff_method="backprop"` on `default.qubit` differentiates through the simulator, so a plain `.to('cpu')` preserves the path. (`lightning.qubit` + `adjoint` is the faster alternative if backprop memory becomes an issue.)

## Related: the v7 per-sample Python loop

```python
results = [self.qlayer(sample) for sample in x_cpu]
```

This runs one QNode execution per element of the batch in a Python loop, on CPU. `TorchLayer` already supports batched inputs (v8 does this correctly — `self.qlayer(x_cpu)` on a `[B, n_qubits]` tensor). Switching v7 to the batched call should be a large wall-clock win and would make the scaling experiments in #5 practical.

## Suggested test

```python
def test_quantum_branch_receives_gradient():
    m = ImprovedUNet(use_quantum=True)
    x = torch.randn(2, 1, 28, 28)
    t = torch.randint(0, 1000, (2,))
    m(x, t).sum().backward()
    assert m.q_attn.input_proj.weight.grad.abs().sum() > 0
    assert m.q_attn.qlayer.weights.grad.abs().sum() > 0
```

Both assertions currently fail (the first with `grad is None`).

**Comment by ChiShengChen** [comment]
Fixed by #7 (merged). GitHub only auto-closes the first issue listed in a `Fixes #1 and #2` line, so this one stayed open.

`QuantumLayer.forward` now does `.to('cpu')` instead of `.detach().cpu()` in all three model files. Verified: the input tensor keeps its gradient and `input_proj` is no longer frozen at initialization — `tests/test_quantum_layer.py::test_input_projection_receives_gradient` covers it.
